### PR TITLE
[dotnet] AppKit pinvokes

### PR DESCRIPTION
**Pull request by stephen-hawley**

### CODE REVIEW

File: src/AppKit/NSApplication.cs [hunk 1]
```diff
@@ -115,7 +120,14 @@ public static void Main (string [] args)
 			// with a call to Main, but this guarantees the right thread.
 			NSApplication.mainThread = Thread.CurrentThread;
 
+#if NET
+			var argsPtr = TransientString.AllocStringArray (args);
+			NSApplicationMain (args.Length, argsPtr);
+			if (argsPtr != IntPtr.Zero)
+				TransientString.FreeStringArray (argsPtr, args.Length);
```
Comment (rolfbjarne) [inline src/AppKit/NSApplication.cs:127]: Can we use this code path for !NET as well?

File: src/AppKit/NSAccessibility.cs [hunk 2]
```diff
@@ -35,11 +39,11 @@ public partial class NSAccessibility {
 		[Mac (10, 10)]
 #endif
 		[DllImport (Constants.AppKitLibrary)]
-		static extern CGRect NSAccessibilityFrameInView (NSView parentView, CGRect frame);
+		static extern CGRect NSAccessibilityFrameInView (NativeHandle parentView, CGRect frame);
 
 		public static CGRect GetFrameInView (NSView parentView, CGRect frame)
 		{
-			return NSAccessibilityFrameInView (parentView, frame);
+			return NSAccessibilityFrameInView (parentView.Handle, frame);
```
Comment (rolfbjarne) [inline src/AppKit/NSAccessibility.cs:46]: This avoids a potential NullReferenceException:

```suggestion
			return NSAccessibilityFrameInView (parentView.GetHandle (), frame);
```

File: src/AppKit/NSAccessibility.cs [hunk 3]
```diff
@@ -48,11 +52,11 @@ public static CGRect GetFrameInView (NSView parentView, CGRect frame)
 		[Mac (10, 10)]
 #endif
 		[DllImport (Constants.AppKitLibrary)]
-		static extern CGPoint NSAccessibilityPointInView (NSView parentView, CGPoint point);
+		static extern CGPoint NSAccessibilityPointInView (NativeHandle parentView, CGPoint point);
 
 		public static CGPoint GetPointInView (NSView parentView, CGPoint point)
 		{
-			return NSAccessibilityPointInView (parentView, point);
+			return NSAccessibilityPointInView (parentView.Handle, point);
```
Comment (rolfbjarne) [inline src/AppKit/NSAccessibility.cs:59]: This avoids a potential NullReferenceException:

```suggestion
			return NSAccessibilityPointInView (parentView.GetHandle (), point);
```

File: src/AppKit/NSGraphics.cs [hunk 4]
```diff
@@ -152,11 +156,11 @@ public static void FrameRect (CGRect rect, nfloat frameWidth, NSCompositingOpera
 		}
 
 		[DllImport (Constants.AppKitLibrary, EntryPoint = "NSShowAnimationEffect")]
-		extern static void NSShowAnimationEffect (nuint animationEffect, CGPoint centerLocation, CGSize size, NSObject animationDelegate, Selector didEndSelector, IntPtr contextInfo);
+		extern static void NSShowAnimationEffect (nuint animationEffect, CGPoint centerLocation, CGSize size, NativeHandle animationDelegate, NativeHandle didEndSelector, IntPtr contextInfo);
 
 		public static void ShowAnimationEffect (NSAnimationEffect animationEffect, CGPoint centerLocation, CGSize size, NSObject animationDelegate, Selector didEndSelector, IntPtr contextInfo)
 		{
-			NSShowAnimationEffect ((nuint) (ulong) animationEffect, centerLocation, size, animationDelegate, didEndSelector, contextInfo);
+			NSShowAnimationEffect ((nuint) (ulong) animationEffect, centerLocation, size, animationDelegate.Handle, didEndSelector.Handle, contextInfo);
```
Comment (rolfbjarne) [inline src/AppKit/NSGraphics.cs:163]: This avoids potential NullReferenceExceptions:

```suggestion
			NSShowAnimationEffect ((nuint) (ulong) animationEffect, centerLocation, size, animationDelegate.GetHandle (), didEndSelector.GetHandle (), contextInfo);
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

**Review by dalexsoto — APPROVED** [review]

**Review by haritha-mohan — APPROVED** [review]

**Review by rolfbjarne — APPROVED** [review]

**Pull request merged**